### ISSUE TITLE
False positives when linting CircleCI Monolith

### ISSUE THREAD

**Issue by marcomorain** [borkdude/clj-kondo#129]
This will be a catch-all issue. I'm not sure how best to isolate repro cases.

1. Wrong number of args passed to function defined in a when-let that shadows the top-level `defn`:

> `foo.clj:13:37: error: wrong number of args (1) passed to user/coercer`

```clojure
(require '[schema.core :as s])

(defn coercer
  "An alternative to `schema.coerce/coercer` that preserves uncoerced originals."
  [schema coercion-matcher]
  (s/start-walker
   (fn [s]
     (let [walker (s/walker s)]
       (if-let [coercer (coercion-matcher s)]
         (fn [x]
           (annotate-error x
                           (try+
                            (let [v (coercer x)] ;; 13:37: error: wrong number of args (1) passed to user/coercer
                              (if (u/error? v)
                                v
                                (walker v)))
                            (catch Error e (throw e))
                            (catch Object t (m/validation-error s x t)))))
         walker)))
   schema))

```

**Comment by borkdude** [comment]
- `13:37: error: wrong number of args (1) passed to user/coercer`
That issue will be solved by this one: https://github.com/borkdude/clj-kondo/issues/111
Coming in the next release.

**Comment by borkdude** [comment]
@marcomorain I just released a new version. Maybe you can verify which errors are still relevant.

**Comment by marcomorain** [comment]
Here is another false positive:

> `/Users/marc/dev/circleci/circle/test/circle/backend/admin/test_db.clj:110:15: error: wrong number of args (2) passed to circle.backend.admin.test-db/stuck-queue-count`

```
(defn- stuck-queue-count
  [calls]
  (or (some->> calls
               first
               :args
               first
               (re-find #"Found (\d+) builds stuck in (?:run|usage)-queue")
               second
               ->int)
      0))


(deftest warn-stuck-picard-builds-works
  (build-queue-fact
    (fn [db builds]
      (letfn [(stuck-queue-count [calls] ...... )]
        ....)))
```

It seems to be a `letfn` shadowing a `defn-` in the same file.

**Comment by borkdude** [comment]
@marcomorain `letfn` isn't properly handled right now by clj-kondo. I'll make a separate issue for that.
as a workaround you can skip linting that macro with:

```
{:linters {:invalid-arity {:skip-args [clojure.core/letfn]}}}
```

**Comment by marcomorain** [comment]
I've replaced the `letfn` with a regular `let`, so all good.

**Comment by borkdude** [comment]
@marcomorain New version released today. That should handle the above issues well now.

**Comment by marcomorain** [comment]
Awesome. I love this tool.
